File: 88-MergeSortedArray/88-MergeSortedArray.java
// Last updated: 4/3/2026, 2:03:13 PM
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if(n==0){}
        else if(m==0) {
            for(int i=0;i<n;i++) {
                nums1[i]=nums2[i];
            }
        }
        else {
            int i=m-1, j=n-1, k=(m+n)-1;
            while(j>=0) {
                if(i>=0 && nums1[i]>nums2[j]) {
                    nums1[k--]=nums1[i--];
                }
                else {
                    nums1[k--]=nums2[j--];
                }
            }
        }
    }
}

File: 88-MergeSortedArray/88-MergeSortedArray.cpp
// Last updated: 4/3/2026, 2:03:05 PM
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int idx = (m + n - 1), i = (m - 1), j(n - 1);
        while (i >= 0 && j >= 0) {
            if (nums1[i] < nums2[j]) {
                nums1[idx] = nums2[j];
                j--;
                idx--;
            } else {
                nums1[idx] = nums1[i];
                i--;
                idx--;
            }
        }
        while (j >= 0) { 
            // When all the elements of nums1 are greater than nums2
            nums1[idx] = nums2[j];
            idx--;
            j--;
        }
    }
};